diff --git a/src/port.h b/src/port.h
index ff18871..75112bb 100644
--- a/src/port.h
+++ b/src/port.h
@@ -86,6 +86,7 @@ extern int errno;  /* Some systems don't declare errno in errno.h. Duh. */
 #if STDC_HEADERS
 # include <stdlib.h>
 # include <string.h>
+# include <strings.h>
 #else
 extern void free();
 # if HAVE_MEMORY_H
diff --git a/src/tfconfig.h.in b/src/tfconfig.h.in
index 56c7076..be26ea2 100644
--- a/src/tfconfig.h.in
+++ b/src/tfconfig.h.in
@@ -12,6 +12,10 @@
  * comment causes problems too.)
  */
 
+#ifndef _DEFAULT_SOURCE
+#define _DEFAULT_SOURCE
+#endif
+
 #define PLATFORM_UNIX 1
 
 #undef MAILDIR
